packages/react-meteor-data/ReactMeteorData.jsx

```diff
@@ -166,4 +122,61 @@ class MeteorDataManager {
   }
 }
 
-export default ReactMeteorData;
+export const ReactMeteorData = {
+  componentWillMount() {
+    this.data = {};
+    this._meteorDataManager = new MeteorDataManager(this);
+    const newData = this._meteorDataManager.calculateData();
+    this._meteorDataManager.updateData(newData);
+  },
+
+  componentWillUpdate(nextProps, nextState) {
+    const saveProps = this.props;
+    const saveState = this.state;
+    let newData;
+    try {
+      // Temporarily assign this.state and this.props,
+      // so that they are seen by getMeteorData!
+      // This is a simulation of how the proposed Observe API
+      // for React will work, which calls observe() after
+      // componentWillUpdate and after props and state are
+      // updated, but before render() is called.
+      // See https://github.com/facebook/react/issues/3398.
+      this.props = nextProps;
+      this.state = nextState;
+      newData = this._meteorDataManager.calculateData();
+    } finally {
+      this.props = saveProps;
+      this.state = saveState;
+    }
+
+    this._meteorDataManager.updateData(newData);
+  },
+
+  componentWillUnmount() {
+    this._meteorDataManager.dispose();
+  },
+};
+
+export function connect({ getMeteorData, pure = true }) {
+  const BaseComponent = pure ? React.PureComponent : React.Component;
```

Comment:
(Read this comment after my below comment) :wink:

Since this may be called very often (for each of a potentially large number of components, no?), might there be a performance gain by defining two intermediary "base" objects (one pure and one non-pure) before the `connect` method and then just doing the single `extend` within the `WrappedComponent` function?